### ISSUE TITLE
Automatically generated TYPO3 AdditionalConfiguration.php overwrites MAIL configuration

### ISSUE THREAD

**Issue by dmnkhhn** [drud/ddev#1042]
**Describe the bug**
The automatically generated TYPO3 AdditionalConfiguration.php overwrites the complete MAIL configuration so that the $GLOBALS['TYPO3_CONF_VARS']['MAIL'] array only contains those two values making it impossible to set or use any other values.

**To Reproduce**
1. Go to 'Configuration' module in the backend
2. Click on the Global Configuration
3. Scroll down to the MAIL section
4. Only two values are present

**Expected behavior**
I would expect a much larger array with the complete MAIL configuration (including defaultMailFromAddress, etc.)

**Screenshots**
<img width="368" alt="bildschirmfoto 2018-08-10 um 21 22 44" src="https://user-images.githubusercontent.com/229676/43977930-dc79c86a-9ce5-11e8-89a1-6c7128f17f61.png">

**Version and configuration information:**
 - macOS Mojave
 - TYPO3 CMS 9.3.3
 - Docker 18.06.0-ce
 - ddev v1.0.0

**Possible solution:**
The part of the file currently looks like this:
```
// This mail configuration sends all emails to mailhog
$GLOBALS['TYPO3_CONF_VARS']['MAIL'] = [
    'transport' => 'smtp',
    'transport_smtp_server' => 'localhost:1025',
];
```
… and it should probably look like this:
```
// This mail configuration sends all emails to mailhog
$GLOBALS['TYPO3_CONF_VARS']['MAIL']['transport'] = 'smtp';
$GLOBALS['TYPO3_CONF_VARS']['MAIL']['transport_smtp_server'] = 'localhost:1025';
```

**Comment by rfay** [comment]
Hmm, this was approved by TYPO3 community and works as is; Normally the user-agent (TYPO3) is responsible for adding headers.

However, if you want to send mail somewhere else or configure it differently, just remove the #ddev-generated comment from the AdditionalConfiguration.php file, and ddev will no longer change it at all. 

And I don't have any objection to your suggestion, but also am pretty sure that the additional expected items in MAIL are not useful for this config:

```
        'transport_smtp_encrypt' => '',
        'transport_smtp_password' => '',
        'transport_smtp_server' => '',
        'transport_smtp_username' => '',
```
None of those are useful.

**Comment by dmnkhhn** [comment]
I agree to a certain extend that of those values you have listed are not useful to change in a ddev environment  but MAIL does have a few other items that are independent from the server settings and thus should be preserved and not overwritten by the AdditionalConfiguration. It kind of beats the purpose of having a AdditionalConfiguration when everything gets overwritten. ;-)

- defaultMailFromAddress
- defaultMailFromName
- defaultMailReplyToAddress
- defaultMailReplyToName

**Comment by rfay** [comment]
Fixed in #1042 - Thanks!
